Agents/DashboardAgent/agent/src/main/java/uk/ac/cam/cares/jps/agent/dashboard/DashboardAgent.java

```diff
@@ -0,0 +1,189 @@
+package uk.ac.cam.cares.jps.agent.dashboard;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+import uk.ac.cam.cares.jps.agent.dashboard.json.DashboardClient;
+import uk.ac.cam.cares.jps.agent.dashboard.stack.StackClient;
+import uk.ac.cam.cares.jps.base.agent.JPSAgent;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class acts as the entry point of the compiled war.
+ *
+ * @author qhouyee
+ */
+@WebServlet(urlPatterns = {"/reset", "/status", "/setup"})
+public class DashboardAgent extends JPSAgent {
+    private static final Logger LOGGER = LogManager.getLogger(DashboardAgent.class);
+    // Agent starts off in valid state, and will be invalid when running into exceptions
+    private static boolean VALID = true;
+    private static StackClient SERVICES;
+    private static final String UNDEFINED_ROUTE_ERROR_MSG = "Invalid route! Requested route does not exist for : ";
+    private static final String INVALID_ROUTE_ERROR_MSG = "Invalid request type! Route ";
+
+    /**
+     * Perform required setup.
+     */
+    @Override
+    public synchronized void init() {
+        try {
+            super.init();
+            // Ensure logging are properly working
+            LOGGER.debug("This is a test DEBUG message");
+            LOGGER.info("This is a test INFO message");
+            LOGGER.warn("This is a test WARN message");
+            LOGGER.error("This is a test ERROR message");
+            LOGGER.fatal("This is a test FATAL message");
+            // Ensure that the agent is running on a stack by initialising the services
+            SERVICES = new StackClient();
+        } catch (ServletException exception) {
+            // This error only occurs when super.init() fails
+            DashboardAgent.VALID = false;
+            LOGGER.error("Could not initialise an agent instance!", exception);
+        } catch (Exception exception) {
+            // This error will only occur if the agent is not running on a stack, which causes an error in Stack Client
+            DashboardAgent.VALID = false;
+            LOGGER.error("Agent is not running on a stack and will not be initialised!", exception);
```

Comment:
See comment above regarding the `VALID` field.
The exceptions should be rethrown or new exceptions that wrap them should be thrown as the agent is in an invalid state.
```suggestion
            DashboardAgent.valid = true;
        } catch (ServletException exception) {
            // This error only occurs when super.init() fails
            LOGGER.error("Could not initialise an agent instance!", exception);
            // Add code to rethrow existing exception or throw a new one that wraps the current one
        } catch (Exception exception) {
            // This error will only occur if the agent is not running on a stack, which causes an error in Stack Client
            LOGGER.error("Agent is not running on a stack and will not be initialised!", exception);
            // Add code to rethrow existing exception or throw a new one that wraps the current one
```